lib/jemoji.rb

```diff
@@ -9,6 +9,7 @@ class Emoji
     GITHUB_DOT_COM_ASSET_HOST_URL = "https://assets-cdn.github.com".freeze
     ASSET_PATH = "/images/icons/".freeze
     BODY_START_TAG = "<body".freeze
+    OPENING_BODY_TAG_REGEX = %r!<body(.*)>\s*!
```

Comment:
1. Does the `.*` need to be non-greedy? (not sure which is more fault tolerant)
2. Does this need to be case insensitive to allow for `<BODY` tags?
3. Could we simplify/collapse the existing logic with `BODY_START_TAG`?

Reply:
Interesting questions @benbalter. I ran a benchmark script to reach a conclusion:
(Results from Ruby 2.3 on Windows)

```ruby
require 'benchmark/ips'

BODY_START_TAG = "<body".freeze
LAZY_MATCH   = %r!#{BODY_START_TAG}(.*?)>\s*!
GREEDY_MATCH = %r!#{BODY_START_TAG}(.*)>\s*!
LAZY_MATCH_MIXCASE   = %r!#{BODY_START_TAG}(.*?)>\s*!i
GREEDY_MATCH_MIXCASE = %r!#{BODY_START_TAG}(.*)>\s*!i

# CONTENT = unmodified HTML output from one of the posts at jekyll/jekyll
# redacted for obvious reasons..

def lazy
  CONTENT.partition(LAZY_MATCH)
end

def greedy
  CONTENT.partition(GREEDY_MATCH)
end

def mixcase_lazy
  CONTENT.partition(LAZY_MATCH_MIXCASE)
end

def mixcase_greedy
  CONTENT.partition(GREEDY_MATCH_MIXCASE)
end

p lazy == greedy
p mixcase_lazy == mixcase_greedy
p lazy == mixcase_greedy

Benchmark.ips do |x|
  x.report('lazy') { lazy }
  x.report('greedy') { greedy }
  x.report('lazy-mixcase') { mixcase_lazy }
  x.report('greedy-mixcase') { mixcase_greedy }
  x.compare!
end
```
```sh
true
true
true
Warming up --------------------------------------
                lazy     5.660k i/100ms
              greedy     5.781k i/100ms
        lazy-mixcase     3.382k i/100ms
      greedy-mixcase     3.465k i/100ms
Calculating -------------------------------------
                lazy     62.413k (A± 3.5%) i/s -    316.960k in   5.085169s
              greedy     64.279k (A± 3.2%) i/s -    323.736k in   5.041942s
        lazy-mixcase     36.254k (A± 1.7%) i/s -    182.628k in   5.038894s
      greedy-mixcase     36.941k (A± 3.2%) i/s -    187.110k in   5.070742s

Comparison:
              greedy:    64278.8 i/s
                lazy:    62412.9 i/s - same-ish: difference falls within error
      greedy-mixcase:    36941.0 i/s - 1.74x  slower
        lazy-mixcase:    36254.5 i/s - 1.77x  slower
```

Therefore, 
  - I'm :-1: on making `(.*)` lazy
  - I'm unsure of modifying `BODY_START_TAG` (a `String`) into a `Regexp`
  - I'm :+1: on case-insensitivity in spite of `%r!<body(.*)>\s*!i` being *slower..* (don't want the plugin to *break* on markup with `<BODY>`)